src/facade/dragonfly_connection.cc

```diff
@@ -769,6 +776,20 @@ auto Connection::IoLoop(util::FiberSocketBase* peer, SinkReplyBuilder* orig_buil
   do {
     FetchBuilderStats(stats_, orig_builder);
 
+    if (migration_request_) {
+      ProactorBase* dest = migration_request_;
+
+      dispatch_fb_.JoinIfNeeded();
+      DCHECK(dispatch_q_.empty());
+      migration_request_ = nullptr;
+      this->Migrate(dest);
+
+      // We're now running in `dest` thread
+      queue_backpressure_ = &tl_queue_backpressure_;
+      dispatch_fb_ =
+          fb2::Fiber(dfly::Launch::post, "connection_dispatch", [&] { DispatchFiber(peer); });
+    }
+
```

Comment:
also take into account that you won't migrate until after the next socket read, which might be too late with a big batch of pipelined commands, but that's a really minor issue.... Just a thought